### DISCUSSION TITLE
Debugger extension, best way to structure  project

### DISCUSSION THREAD

**Discussion by trixtan** [microsoft/vscode-discussions#300] [Extension Development QnA] [answered]
Hello,

I am writing a debugger extension for the Infineon AURIX microcontrollers. My idea is to create a C++ DAP adapter around their MCD API.

I read the VSCode documentation, and I know I have to create an extension project to interface the DA with VSCode. The extension will also be responsible for implementing the debugger configuration commands (device selection, etc)

Now my question: how should I structure my project? My first idea was to have a C++ project for the DA, and one for the extension. Is this a good way? Can I have a single project supporting C++ and typescript, building and debugging both?

Thank you,
Kind regards
Enrico

**Comment by isidorn** [comment, marked as answer]
@trixtan I think this is personal preference in how you would structure your project. And yes, VS Code supports debugging multiple languages in a single project - like Typescript and C++.

For inspiration I suggest to checkout existing Debug Adapter implementations and how they have structured their projects. Here's the list https://microsoft.github.io/debug-adapter-protocol/implementors/adapters/

**Comment by trixtan** [comment]
Thank you for the hint!

Il giorno mer 14 dic 2022 alle ore 16:38 Isidor Nikolic <
***@***.***> ha scritto:
